### PR TITLE
Ubuntu14.04/v1.1

### PR DESCRIPTION
**Pull request by themangoemoji**
@mr-c 
Is over here better?
I'll go ahead and make the pip install change.

### CODE REVIEW

**Comment by mr-c** [comment]
Yep, this is the correct repository :-)

**Comment by mr-c** [comment]
Looking pretty good so far, let me know when you're ready for a review

**Comment by mr-c** [comment]
As we discussed let removed all references to KHMER in the makefile and test that the clean-list.txt is up to date by running `make all && make clean && git status`

**Comment by themangoemoji** [comment]
- [x] Is it mergable
- N/A Did it pass the tests?
- N/A If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `pep8`/`pylint`, `cppcheck`, and
  `make doc` output. Use `autopep8` and `astyle -A10 --max-code-length=80`
  if needed.
- [ ] Is it documented in the Changelog?
- [ ] Was spellcheck run on the source code and documentation after changes
  were made?

File: README.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
 Source code and analysis pipeline for
 
 http://ged.msu.edu/papers/2012-diginorm/
+
+------------------------------------------
+
+Quick look at ipython notebook results at:
+
+http://nbviewer.ipython.org/github/wrightmhw/ipynb_temp/blob/master/diginorm.ipynb
```
Comment (mr-c) [inline README.txt]: Awesome. @ctb do you want this here or in another location?

File: pipeline/pipeline-notes.rst [hunk 2]
```diff
@@ -24,63 +24,42 @@ below software installed.)
...
 
- cd /usr/local/share
 
- git clone https://github.com/ged-lab/screed.git
- cd screed
- git checkout 2012-paper-diginorm
- python setup.py install
- cd ..
+We will need pip, ipython, and the ipython notebook.  Make sure we have 
+the latest version of ipython notebook (you need 0.13dev, or later) ::
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: If there is a version requirement you can specify it below in the apt-get install line.

File: pipeline/pipeline-notes.rst [hunk 3]
```diff
@@ -24,63 +24,42 @@ below software installed.)
...
 
- git clone https://github.com/ged-lab/khmer.git
- cd khmer
- git checkout 2012-paper-diginorm
- make test
- cd ..
+ apt-get install python-pip
+ apt-get install ipython
+ apt-get install ipython-notebook
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: Should these be combined with the initial apt-get line at the beginning of this document?

File: pipeline/pipeline-notes.rst [hunk 4]
```diff
@@ -24,63 +24,42 @@ below software installed.)
...
 
- echo export PYTHONPATH=/usr/local/share/khmer >> ~/.bashrc
- echo 'export PATH=$PATH:/usr/local/share/khmer/scripts' >> ~/.bashrc
- echo 'export PATH=$PATH:/usr/local/share/khmer/sandbox' >> ~/.bashrc
- source ~/.bashrc
+Now, you'll need to install the version of 'khmer' that the
+paper is currently using.::
+ 
+ easy_install -U setuptools
+ pip install khmer==1.1,
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: trailing comma

**Comment by mr-c** [comment]
This looks good. @ctb will want to do a final review

@ctb Should this be tested with literate-resting?

**Comment by themangoemoji** [comment]
Michael, The nbdiff shows little more than label differences (between which I do not see any distinctions). Also, there is no support for output for the tool. The diff is hosted locally on the browser.

**Comment by mr-c** [comment]
Can you pipe the output to a text file? `nbdiff $OPTIONS |& tee output.txt`
should do the trick.

**Comment by themangoemoji** [comment]
Hmmm seems that this will not pipe out to txt files after all. I will look into more. there doesn't seem to be any documentation on it though..

**Comment by mr-c** [comment]
Were you able to save the nbdiff webpage?

**Comment by themangoemoji** [comment]
Yep yep. it is in the pull request. I'm going to try to make sure it saves
with the pretty figures though.

On Tue, Jul 15, 2014 at 3:20 PM, Michael R. Crusoe <notifications@github.com

**Comment by mr-c** [comment]
Nifty. A link to the diff would be easiest to review. The diff files
shouldn't go inside the pull request because we aren't going to include it
in the repository. Maybe host it in your Athyra web space?

**Comment by themangoemoji** [comment]
http://athyra.ged.msu.edu/~wrightm/nbdiff.html

hmmm I need to figure out the permissions on hosting the nbdiff_files/

**Comment by mr-c** [comment]
Bummer. Use
https://chrome.google.com/webstore/detail/webpage-screenshot/ckibcdccnfeookdmbahgiakhnjcddpki?hl=en
to take a picture?

**Comment by themangoemoji** [comment]
Yeah luiz just suggested that too.

**Comment by themangoemoji** [comment]
@ctb 
The final issue with the diginorm pipeline (runs smoothly aside from this): 
It appears that blast has changed the way they deal with file extensions/compatibility. `sar_dn.faa` needs to be `sar_dn.fa`. I believe this comes from:
`https://s3.amazonaws.com/public.ged.msu.edu/2012-paper-diginorm/pipeline-data-new.tar.gz`
Can I change this file extension from the source or should I change it within the pipeline?

**Comment by ctb** [comment]
On Wed, Jul 23, 2014 at 07:07:01PM -0700, Michael Wright wrote:

> @ctb 
> The final issue with the diginorm pipeline (runs smoothly aside from this): 
> It appears that blast has changed the way they deal with file extensions/compatibility. 'sar_dn.faa' needs to be 'sar_dn.fa'. I believe this comes from:
> 'https://s3.amazonaws.com/public.ged.msu.edu/2012-paper-diginorm/pipeline-data-new.tar.gz'
> Can I change this file extension from the source or should I change it within the pipeline?

I highly doubt that .faa change is needed, but if it makes your life easier,
please change w/in pipeline.

## --t

C. Titus Brown, ctb@msu.edu

File: README.txt [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+
+
+nbdiff of pipeline run on old and new khmer versions:
+
+http://athyra.ged.msu.edu/~wrightm/nbdiff_image.png
```
Comment (mr-c) [inline README.txt]: Ah, I just saw this. To ensure that your reviewers see notes like this put them in a comment so they get a notification. They shouldn't go here.

**Comment by themangoemoji** [comment]
Quick look at ipython notebook results at:
http://nbviewer.ipython.org/github/wrightmhw/ipynb_temp/blob/master/diginorm.ipynb

nbdiff of pipeline run on old and new khmer versions:
http://athyra.ged.msu.edu/~wrightm/nbdiff_image.png

**Comment by mr-c** [comment]
Looks great, thanks!

File: pipeline/Makefile [hunk 6]
```diff
@@ -82,62 +80,62 @@ endbias-transcripts.txt: transcript-reads.fa.keep.abundfilt.keep.kh
 	python find-missing-kmers.py transcript-reads.fa.keep.abundfilt.keep.kh transcripts.fa endbias-transcripts.txt
 
 genome-reads.fa.kh: genome-reads.fa
-	${KHMER}/scripts/load-into-counting.py -k 20 -N 4 -x 1e8 genome-reads.fa.kh genome-reads.fa
+	load-into-counting.py -k 20 -N 4 -x 1e8 genome-reads.fa.kh genome-reads.fa
```
Comment (mr-c) [inline pipeline/Makefile]: As part of the move to v1.1 we need to change the file extension from ".kh" to ".pt" or ".ct" as appropriate
Counting tables such as produced by load-into-counting should have the extension ".ct"

When you're done the string ".kh" shouldn't appear in the Makefile anymore

File: pipeline/Makefile [hunk 7]
```diff
@@ -364,7 +362,7 @@ sar_dn.faa: sar.keep.assembly.fa
 
 staph_dn.faa: staph.keep.assembly.fa
 	/mnt/prokka-1.7/bin/prokka staph.keep.assembly.fa --outdir staph_dn --prefix staph_dn --force
-	cp staph_dn/staph_dn.faa .
+	cp staph_dn/staph_dn.fa .
```
Comment (mr-c) [inline pipeline/Makefile]: This change needs to be reverted

File: pipeline/pipeline-notes.rst [hunk 8]
```diff
@@ -2,7 +2,7 @@
 Running the diginorm paper script pipeline
 ==========================================
 
-:Date: Oct 28, 2013
+:Date: Jun 25, 2014
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: Please update this date :-)

File: pipeline/pipeline-notes.rst [hunk 9]
```diff
@@ -10,77 +10,43 @@ normalization on an Amazon EC2 rental instance.
...
+ sudo su
+ mkfs.ext4 /dev/xvdb
+ mount -t ext4 /dev/xvdb /mnt/
 
- git clone https://github.com/ged-lab/screed.git
- cd screed
- git checkout 2012-paper-diginorm
- python setup.py install
- cd ..
+Just ssh in however you would normally do it. And then set the instance up
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: The user was already told to SSH in.

File: pipeline/pipeline-notes.rst [hunk 10]
```diff
@@ -10,77 +10,43 @@ normalization on an Amazon EC2 rental instance.
...
+with all of the software we will need::
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: all of the software -> many of the software packages

File: pipeline/pipeline-notes.rst [hunk 11]
```diff
@@ -89,11 +55,6 @@ and Velvet::
  make MAXKMERLENGTH=51
  cp velvet? /usr/local/bin
 
-Finally, upgrade the latex install with a few recommended packages, and
-add mummer::
-
- apt-get install -y texlive-latex-recommended mummer
-
 OK, now all your software is installed, hurrah!
```
Comment (mr-c) [inline pipeline/pipeline-notes.rst]: This line is a lie (see Prokka below)

**Comment by mr-c** [comment]
Address my comments and I think you're done. @ctb What do you think?

**Comment by mr-c** [comment]
Please remove the file "notebook/diginorm_old.ipynb"

**Comment by themangoemoji** [comment]
How does this look, @ctb?

File: pipeline/Makefile [hunk 12]
```diff
@@ -51,10 +49,10 @@ copydata:
...
 	rm -f `cat clean-list.txt`
 
-cleankh:
-	rm -f bacteria_mda_lane1.fastq.bz2.normC20k20.kh \
-		bacteria_mda_lane7.fastq.bz2.normC20k20.kh \
-		ecoli_ref.fastq.bz2.normC20k20.kh
+cleanpt:
+	rm -f bacteria_mda_lane1.fastq.bz2.normC20k20.ct \
+		bacteria_mda_lane7.fastq.bz2.normC20k20.ct \
+		ecoli_ref.fastq.bz2.normC20k20.ct
```
Comment (mr-c) [inline pipeline/Makefile:55]: counting tables here, but makefile target is 'cleanpt' ?

**Comment by mr-c** [comment]
Please remove `KHMER=/usr/local/share/khmer` from pipeline readme

**Comment by b-wyss** [comment]
Followed instructions to make the paper, as detailed at https://github.com/ged-lab/2012-paper-diginorm/blob/ubuntu14.04/v1.1/pipeline/pipeline-notes.rst, with the change (per @mr-c) of using 

```
make
```

rather than 

```
make /usr/local/share/khmer
```

and got the following error: 

```
make: *** No rule to make target `ecoli_ref-5m.fastq', needed by `ecoli_ref-5m.fastq.map'.  Stop.
```

**Comment by themangoemoji** [comment]
I believe I fixed this, Brian. I was copying velvet? to ~/bin and not {HOME}/bin. In the amazon instance at least, this is usr/ubuntu/bin. I am getting hung up later in this make process. Trying to sort that out now and then maybe you can rerun?

**Comment by mr-c** [comment]
@wrightmhw  `~/bin` and `${HOME}/bin` should be the same thing.

I don't know what you mean by 'usr/ubuntu/bin'.

File: .profile [hunk 13]
```diff
@@ -0,0 +1,3 @@
+if [ -d "$HOME/bin" ] ; then
```
Comment (mr-c) [inline .profile:1]: @wrightmhw why did you include this file?

**Comment by mr-c** [comment]
@wrightmhw also @b-wyss was testing the `ubuntu14.04/v1.1` branch and not the `non-root-install` branch

**Comment by mr-c** [comment]
@ctb and @camillescott Can you review this?

**Comment by mr-c** [comment]
Prodigal is now apt-get installable: http://packages.ubuntu.com/search?keywords=prodigal

**Pull request closed without merging**